plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -533,7 +543,16 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
     public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
     {
         try {
-            return createTable(session, tableMetadata, generateTemporaryTableName());
+            JdbcOutputTableHandle handle = createTable(session, tableMetadata, generateTemporaryTableName());
+            if (includePageSinkIdColumn(session)) {
+                // TODO: Should we clean up the table if this fails?
+                addColumn(session, new RemoteTableName(
```

Comment:
another approach would be to pass "extracolumns" to `createTable` call. But I think all the databases support adding columns so this should be fine.

Reply:
🤔 yeah, i actually think that's a better approach.

Reply:
just pushed code that makes this change